test/integ/pandas/test_pandas_tools.py

```diff
@@ -340,10 +351,44 @@ def test_special_name_quoting(
             # Check table's contents
             result = cnx.cursor(DictCursor).execute(select_sql).fetchall()
             for row in result:
-                assert row["id"] in (1, 2)
+                # The auto create table functionality does not auto-create an incrementing ID
+                if not auto_create_table:
+                    assert row["id"] in (1, 2)
                 assert (
                     row["00name"],
                     row["bAlance"],
                 ) in df_data
         finally:
             cnx.execute_string(drop_sql)
+
+
+def test_auto_create_table_similar_column_names(
+    conn_cnx: Callable[..., Generator["SnowflakeConnection", None, None]],
+):
+    """Tests whether similar names cause issues when auto-creating a table as expected."""
+    table_name = "numbas"
```

Comment:
Use name randomizer here, please!

Reply:
Is there a name randomizer for tests?

Reply:
Yes, here's an example: https://github.com/snowflakedb/snowflake-connector-python/blob/21cd719d6b7f599d1da1792a65212668ce482535/test/integ/test_put_get_user_stage.py#L142